manim/mobject/mobject.py

```diff
@@ -1437,7 +1455,7 @@ def construct(self):
                     self.play(t.animate.set_value(TAU), run_time=3)
         """
 
-        def R3_func(point):
+        def R3_func(point: Point3D) -> Point3D:
             x, y, z = point
             xy_complex = function(complex(x, y))
             return [xy_complex.real, xy_complex.imag, z]
```

Comment:
Although `R3_func()` is typed to return a `Point3D`, it is returning a list which actually corresponds to a `Point3DLike`.

After some thoughts, I realized that maybe typing these methods as accepting a `MappingFunction` (a `Callable[[Point3D], Point3D]`) is a bit too strict. They technically can accept a `Callable[[Point3DLike], Point3DLike]`. However, that is better suited for a follow-up discussion and possible PR.

In the meantime, let's just make the return type a NumPy array:

```suggestion
        def R3_func(point: Point3D) -> Point3D:
            x, y, z = point
            xy_complex = function(complex(x, y))
            return np.array([xy_complex.real, xy_complex.imag, z])
```